Automattic/jetpack
# Search: allow only valid post types in Customberg

## Proposed changes:

If two or more post types are set and then these post types got removed (because of plugin removed etc), the excluded post type option would become disabled. The PR proposes to filter on the invalid post types, i.e. remove out-dated settings.

### Other information:


## Jetpack product discussion
panfyZ-2PW-p2


## Does this pull request change what data or activity we track or use?
no

## Testing instructions:

* Build search and Jetpack locally `jetpack build packages/search && jetpack build plugins/jetpack`
* Update value of `jetpack_search_excluded_post_types` option to `aaa,bbb`
* Open `/wp-admin/admin.php?page=jetpack-search-configure`
* Ensure Excluded post types setting is available for changes
* Ensure you excluded post types stick after click saving/page refresh

<img width="184" alt="image" src="https://github.com/Automattic/jetpack/assets/1425433/2183c1d1-b7b1-41cf-b1fa-981da07e658d">
